build_defs/kt_opts.bzl

```diff
@@ -0,0 +1,96 @@
+"""Inlined version of `kt_jvm_export` from `rules_jvm_external`."""
+
+load("@rules_kotlin//kotlin:jvm.bzl", "kt_jvm_library")
+load("@rules_jvm_external//private/rules:java_export.bzl", "maven_export")
+load("@rules_jvm_external//private/rules:maven_project_jar.bzl", "DEFAULT_EXCLUDED_WORKSPACES")
+load("//build_defs:java_opts.bzl", "DEFAULT_MANIFEST_ENTRIES")
+
+def kt_jvm_export(
+        name,
+        maven_coordinates,
+        manifest_entries = DEFAULT_MANIFEST_ENTRIES,
+        deploy_env = [],
+        excluded_workspaces = {name: None for name in DEFAULT_EXCLUDED_WORKSPACES},
+        pom_template = None,
+        visibility = None,
+        tags = [],
+        testonly = None,
+        **kwargs):
+    """Extends `kt_jvm_library` to allow maven artifacts to be uploaded. This
+    rule is the Kotlin JVM version of `java_export`.
+
+    This macro can be used as a drop-in replacement for `kt_jvm_library`, but
+    also generates an implicit `name.publish` target that can be run to publish
+    maven artifacts derived from this macro to a maven repository. The publish
+    rule understands the following variables (declared using `--define` when
+    using `bazel run`):
```

Comment:
`kt_jvm_export` inlined from [here](https://github.com/bazelbuild/rules_jvm_external/blob/master/private/rules/kt_jvm_export.bzl) in order to support `manifest_entries = { ... }`, which the underlying [`java_export`](https://github.com/bazelbuild/rules_jvm_external/pull/1050) rule supports, but is not exposed yet and not passable via `**kwargs`. I'll make sure to offer a PR and issue upstream for this.